relay-wstring/src/lib.rs

```diff
@@ -262,18 +262,23 @@ impl<'a> Iterator for WStrChars<'a> {
     fn next(&mut self) -> Option<Self::Item> {
         // Our input is valid UTF-16LE, so we can take a lot of shortcuts.
         let chunk = self.chunks.next()?;
-        let u = u16::copy_from_le_bytes(chunk)?;
+        // Chunks always returns 2 bytes, avoid generating panicking code
+        let u = u16::copy_from_le_bytes(chunk).unwrap_or_default();
 
-        if !is_leading_surrogate(u) {
+        if is_standalone_unit(u) {
             // SAFETY: This is now guaranteed a valid Unicode code point.
-            Some(unsafe { std::char::from_u32_unchecked(u as u32) })
+            Some(unsafe { std::char::from_u32_unchecked(u.into()) })
         } else {
+            debug_assert!(is_leading_surrogate(u), "code unit not a leading surrogate");
+
             let chunk = self.chunks.next().expect("missing trailing surrogate");
```

Comment:
we still have a potential panic here. Are we *absolutely sure* that we always get two matching surrogate pairs? AFAICR, unmatched surrogates is a quite common thing when dealing with utf-16.

Or is that case covered somewhere else?

Reply:
Yeah, we were in two minds here.  These `.expects()` in these places just saved us by notifying us of this bug...

The reasoning to let this one is because this expects non-local code, where the other panic-avoiding code is super local and can be reasoned about locally.  Still, maybe that's the wrong call and we should avoid panicing code entirely.

Wrt the surrogate pair.  The whole premise of `WStr` is that it absolutely ensures you always have valid UTF-16.  And that means no lone surrogates.